### PR TITLE
Delete AvailableResources judgment for GA features

### PR DESCRIPTION
**Pull request by chenyw1990**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

 Just like the PR [Graduate EndpointSlice Controllers to GA](https://github.com/kubernetes/kubernetes/pull/99870), we should delete the AvailableResources judgment in startXXXController for GA features.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kube-controller-manager/app/rbac.go [hunk 1]
```diff
@@ -19,14 +19,10 @@ package app
 import (
 	"net/http"
 
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/kubernetes/pkg/controller/clusterroleaggregation"
 )
 
 func startClusterRoleAggregrationController(ctx ControllerContext) (http.Handler, bool, error) {
-	if !ctx.AvailableResources[schema.GroupVersionResource{Group: "rbac.authorization.k8s.io", Version: "v1", Resource: "clusterroles"}] {
```
Comment (deads2k) [inline cmd/kube-controller-manager/app/rbac.go:27]: I don't think the RBAC APIs are required for conformance, can you link me?

**Comment by deads2k** [comment]
> Just like the PR [Graduate EndpointSlice Controllers to GA](https://github.com/kubernetes/kubernetes/pull/99870), we should delete the AvailableResources judgment in startXXXController for GA features.

I don't  think that GA is a sufficiently high bar to remove these checks. I think the bar is conformance.  It is possible to have clusters which are conformant kube clusters, but do not have all GA APIs enabled.  I'm pretty sure that autoscaling and RBAC are not required by conformance, but if you find their test, link me.

You can also limit scope to just those APIs required for conformance and I'll approve.

File: cmd/kube-controller-manager/app/autoscaling.go [hunk 2]
```diff
@@ -35,10 +34,6 @@ import (
 )
 
 func startHPAController(ctx ControllerContext) (http.Handler, bool, error) {
-	if !ctx.AvailableResources[schema.GroupVersionResource{Group: "autoscaling", Version: "v1", Resource: "horizontalpodautoscalers"}] {
```
[the remark this answers is not in the document]
Comment (chenyw1990) [reply cmd/kube-controller-manager/app/autoscaling.go]: I have added this judgment back

File: cmd/kube-controller-manager/app/rbac.go (hunk 1 above)
Comment (chenyw1990) [reply cmd/kube-controller-manager/app/rbac.go]: I have added this judgment back

**Comment by chenyw1990** [comment]
I have modified the code based on the review comments, and delete the AvailableResources condition in startCronJobController

File: cmd/kube-controller-manager/app/batch.go [hunk 3]
```diff
@@ -44,9 +40,6 @@ func startJobController(ctx ControllerContext) (http.Handler, bool, error) {
 }
 
 func startCronJobController(ctx ControllerContext) (http.Handler, bool, error) {
-	if !ctx.AvailableResources[schema.GroupVersionResource{Group: "batch", Version: "v1", Resource: "cronjobs"}] {
```
Comment (deads2k) [inline cmd/kube-controller-manager/app/batch.go:47]: Help me find the cronjob conformance test?  it's GA, but I don't see any `framework.ConformanceIt` test for cronjobs.

Comment (deads2k) [reply cmd/kube-controller-manager/app/batch.go:47]: > Help me find the cronjob conformance test? it's GA, but I don't see any `framework.ConformanceIt` test for cronjobs.

nevermind, stale level.  I found it.

**Pull request merged**